source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -288,30 +325,31 @@ The SDAM specification defines `when
 <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#connection-pool-creation>`_
 the driver should create connection pools.
 
-Once a pool is created, if minPoolSize is set, the pool MUST immediately
-begin creating enough connections such that totalConnections >= minPoolSize.
-These connections MUST be created in a non-blocking manner. The driver SHOULD
-set up (connect, handshake and authenticate) these connections in a non-blocking
-manner, such as via the use of a background thread or asynchronous I/O.
+Once a pool is created, if minPoolSize is set, the pool MUST
+immediately begin populating enough `Connections <#connection>`_ such that
+totalConnections >= minPoolSize. These `Connections <#connection>`_ MUST be
+populated in a non-blocking manner, such as via the use of a
+background thread or asynchronous I/O. See `Populating the Pool with a
+Connection <#populating-the-pool-with-a-connection-internal-implementation>`_ for more details.
 
 .. code::
 
     set generation to 0
     emit PoolCreatedEvent
     if minPoolSize is set:
+      # this MAY be performed on a background thread
+      # if it is not performed on a background thread, this MUST
+      # utilize non-blocking I/O.
       while totalConnectionCount < minPoolSize:
```

Comment:
I suggest changing this to `while total count <= max pool size and available + in use count < min pool size`.